container/new_push.bzl

```diff
@@ -59,65 +59,45 @@ def _impl(ctx):
-        pusher_args += ["-stampInfoFile", "%s" % _get_runfile_path(ctx, f)]
-
-    # Find and set src to correct paths depending the image format to be pushed
-    if ctx.attr.format == "oci":
-        found = False
-        for f in ctx.files.image:
-            digester_input.append(f)
-            if f.basename == "index.json":
-                pusher_args += ["-src", "{index_dir}".format(
-                    index_dir = _get_runfile_path(ctx, f),
-                )]
-                digester_args += ["-src", str(f.path)]
-                found = True
-        if not found:
-            fail("Did not find an index.json in the image attribute {} specified to {}".format(ctx.attr.image, ctx.label))
-    if ctx.attr.format == "docker":
-        if len(ctx.files.image) == 0:
-            fail("Attribute image {} to {} did not contain an image tarball".format(ctx.attr.image, ctx.label))
-        if len(ctx.files.image) > 1:
-            fail("Attribute image {} to {} had {} files. Expected exactly 1".format(ctx.attr.image, ctx.label, len(ctx.files.image)))
-        pusher_args += ["-src", _get_runfile_path(ctx, ctx.files.image[0])]
-        digester_args += ["-src", ctx.files.image[0].path]
-        digester_input = [ctx.files.image[0]]
-    if ctx.attr.format == "legacy":
-        # Construct container_parts for input to pusher.
-        image = _get_layers(ctx, ctx.label.name, ctx.attr.image)
-        blobs = image.get("zipped_layer", [])
-        config = image["config"]
-        legacy_files = [config] + blobs
-        tarball = image.get("legacy")
-        if tarball:
-            print("Pushing an image based on a tarball can be very " +
-                  "expensive.  If the image is the output of a " +
-                  "docker_build, consider dropping the '.tar' extension. " +
-                  "If the image is checked in, consider using " +
-                  "docker_import instead.")
-            pusher_args += ["-legacyBaseImage", "%s" % _get_runfile_path(ctx, tarball)]
-            digester_args += ["-legacyBaseImage", "%s" % tarball.path]
-            legacy_files += [tarball]
-
-        pusher_args += ["-configPath", "{}".format(_get_runfile_path(ctx, config))]
-        digester_args += ["-configPath", str(config.path)]
-        digester_input = legacy_files
+        pusher_args += ["-stamp-info-file", "%s" % _get_runfile_path(ctx, f)]
+    pusher_input += stamp_inputs
+
+    # Construct container_parts for input to pusher.
+    image = _get_layers(ctx, ctx.label.name, ctx.attr.image)
+    blobs = image.get("zipped_layer", [])
+    config = image["config"]
+    legacy_files = [config] + blobs
+    pusher_input += legacy_files
+    digester_input += legacy_files
+    tarball = image.get("legacy")
+    if tarball:
+        print("Pushing an image based on a tarball can be very " +
+              "expensive.  If the image is the output of a " +
+              "docker_build, consider dropping the '.tar' extension. " +
```

Comment:
update this doc (docker_build is an alias for container_image)

Reply:
Done